### PR TITLE
Add Destroy callback for squirrel scripts.

### PR DESCRIPTION
**Pull request by Erlite**
This PR adds a new callback to configure in a mod's `mod.json` file. The `Destroy` callback will be executed right before the VM of the given context is destroyed:

```json
{
    "Path": "spyglass/client_globals.nut",
    "RunOn": "CLIENT && MP",
    "ClientCallback":
    {
        "After": "Spyglass_ClientGlobalsInit",
        "Destroy": "Spyglass_ClientGlobalsDestroy"
    }
}
```
```lua
void function Spyglass_ClientGlobalsDestroy()
{
    printt("hello i get destroyed oh no")
}
```

The above example would give the following result:

image

### CODE REVIEW

File: NorthstarDLL/squirrel/squirrel.cpp [hunk 1]
```diff
@@ -209,7 +210,47 @@ template <ScriptContext context> void SquirrelManager<context>::VMCreated(CSquir
...
+					{
+						NS::log::squirrel_logger<context>()->error(
+							"VMDestroyed() was unable to find function with name '{}'. Is it global?", callback.DestroyCallback);
+						continue;
+					}
+
+					pushobject(m_pSQVM->sqvm, &functionobj);
+					pushroottable(m_pSQVM->sqvm);
+
+					_call(m_pSQVM->sqvm, 0);
```
Comment (emma-miler) [inline NorthstarDLL/squirrel/squirrel.cpp:241]: Why do you manually use `_call` here, as opposed to using `Call`?

Comment (Erlite) [reply NorthstarDLL/squirrel/squirrel.cpp:241]: I just copied code from AsyncCall() tbh heh. I'll fix it.

**Comment by BobTheBob9** [comment]
Waiting on changes mentioned in Emma's previous comment

**Review by BobTheBob9 — APPROVED** [review]
compiles and works fine in testing

**Pull request merged**